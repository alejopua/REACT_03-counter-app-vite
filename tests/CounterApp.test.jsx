const { render, screen, fireEvent } = require("@testing-library/react");
const { CounterApp } = require("../src/CounterApp");

describe('Testing in <CounterApp />', () => {
  const value = 0;

  test('must match with the snapshot', () => { 
    const { container } = render( <CounterApp value={ value } /> );
    expect( container ).toMatchSnapshot();
  });

  test('must be increase with the button at +1', () => { 
    render( <CounterApp value={ value } /> );
    fireEvent.click( screen.getByText('+1'));
    expect(screen.getByTestId('counter-value').textContent).toContain('1');
  });

  test('must decrease with the button at -1', () => { 
    render( <CounterApp value={ value } /> );
    fireEvent.click( screen.getByText('-1'));
    // screen.debug();
    expect(screen.getByTestId('counter-value').textContent).toContain('-1');
  });

  test('must be reset to 0', () => { 
    render( <CounterApp value={ value } /> );
    fireEvent.click( screen.getByText('Reset'));
    expect(screen.getByTestId('counter-value').textContent).toContain('0');
  });

});