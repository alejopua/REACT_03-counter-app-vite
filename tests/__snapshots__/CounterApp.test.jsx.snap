// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <CounterApp /> must match with the snapshot 1`] = `
<div>
  <div
    class="h-screen flex flex-col justify-center items-center"
  >
    <h1
      class="text-6xl font-bold text-teal-50"
    >
      Counter App
    </h1>
    <h2
      class="text-4xl font-bold text-red-500"
      data-testid="counter-value"
    >
       
      0
       
    </h2>
    <div
      class="flex bg-[#778da9] mt-1 p-3 rounded-md gap-x-2"
    >
      <button
        class="bg-slate-400 px-6 py-1 border-solid border-4 border-black rounded-md text-3xl font-bold"
        value="0"
      >
         -1 
      </button>
      <button
        class="bg-slate-400 px-6 py-1 border-solid border-4 border-black rounded-md text-3xl font-bold"
        value="0"
      >
         Reset 
      </button>
      <button
        class="bg-slate-400 px-6 py-1 border-solid border-4 border-black rounded-md text-3xl font-bold"
        value="0"
      >
         +1 
      </button>
    </div>
  </div>
</div>
`;
